crates/polars-arrow/src/io/ipc/read/common.rs

```diff
@@ -101,26 +100,6 @@ pub fn read_record_batch<R: Read + Seek>(
         .unwrap_or_else(VecDeque::new);
     let mut buffers: VecDeque<arrow_format::ipc::BufferRef> = buffers.iter().collect();
 
-    // check that the sum of the sizes of all buffers is <= than the size of the file
```

Comment:
Can you explain why this check was removed?

Reply:
I can't easily get file_size after removing the intermediate vec. But remove this check will indeed cause wrong result when file_size and buffer_size not match rather then report an error. Maybe we can change all the subsequent `read_to_end` to `read_excat`?

Reply:
@orlp now all subsequent `read_to_end` has been change to `read_excat`. `read_excat` require vec initial overhead, but I could try to eliminate this using `read_buf_excat` in another pr, if unstable api is acceptable.

Reply:
@orlp done